pkgs/development/interpreters/bats/default.nix

```diff
@@ -1,24 +1,32 @@
-{ stdenv, fetchzip, coreutils, gnugrep }:
+{ stdenv, lib, fetchzip, bash, makeWrapper, coreutils, gnugrep }:
 
 stdenv.mkDerivation rec {
   pname = "bats";
-  version = "1.2.0";
+  version = "1.2.1";
 
   src = fetchzip {
     url = "https://github.com/bats-core/bats-core/archive/v${version}.tar.gz";
-    sha256 = "0f59zh4d4pa1a7ybs5zl6h0csbqqv11lbnq0jl1dgwm1s6p49bsq";
+    hash = "sha256-grB/rJaDU0fuw4Hm3/9nI2px8KZnSWqRjTJPd7Mmb7s=";
   };
 
+  nativeBuildInputs = [ makeWrapper ];
+
   patchPhase = ''
-    patchShebangs ./install.sh
+    patchShebangs .
+  '';
 
-    substituteInPlace ./libexec/bats-core/bats \
-        --replace 'type -p greadlink readlink' 'type -p ${coreutils}/bin/readlink'
-    substituteInPlace ./libexec/bats-core/bats-format-tap-stream \
-        --replace grep ${gnugrep}/bin/grep
+  installPhase = ''
+    ./install.sh $out
+    wrapProgram $out/bin/bats --suffix PATH : "${lib.makeBinPath [ bash coreutils gnugrep ]}"
```

Comment:
After cleaning up the substitutes that didn't apply, I noticed there were still a fair number of grep/coreutils invocations around in other places.

Rather than try and find all of these (resholve might be nice for this, but resholve uses bats for its own tests...) I decided to just wrap it. Do you think that'll cause trouble?

Reply:
Only thing I can think about are shebangs on darwin.

Reply:
If I'm interpreting right, this is referring to a restriction on using a shell script as a shebang. bats is already a shell script, so I don't think it would behave any differently than it already does.